src/suspendable/core.clj

```diff
@@ -25,8 +25,16 @@
 (defn- stop-missing-components [old-system new-system]
   (component/update-system old-system (missing-keys old-system new-system) component/stop))
 
+(defn- object? [x]
+  (instance? clojure.lang.IObj x))
+
+(defn assoc-meta [component key]
+  (if (object? component)
+    (vary-meta component assoc ::key key)
+    component))
+
 (defn- assoc-component-key [system key]
-  (update-in system [key] vary-meta assoc ::key key))
+  (update-in system [key] assoc-meta key))
```

Comment:
`maybe-vary-meta` is exactly what I wrote initially. I changed it because I felt that I was doing too much in a single function, but I agree `object?` is ambiguous given that we're working with `java.lang.Object`s all around us.

I couldn't use `IObj` without the `clojure.lang` prefix so went for the fully qualified version.